clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -200,6 +200,41 @@ bool Sema::isKnownGoodSYCLDecl(const Decl *D) {
   return false;
 }
 
+bool isZeroSizedArray(QualType Ty) {
+  if (const auto *CATy = dyn_cast<ConstantArrayType>(Ty)) {
+    return (CATy->getSize() == 0);
+  }
+  return false;
+}
+
+void Sema::checkSYCLVarDeclIfInKernel(VarDecl *Var) {
+  // not all variable types supported in kernel contexts
+  // if not we record a deferred diagnostic.
```

Comment:
I don't get what you wanted to say by this comment.

Reply:
moved it and revised it a bit.